ide/src/main/java/com/devonfw/tools/ide/url/updater/AbstractUrlUpdater.java

```diff
@@ -257,18 +254,27 @@ private static boolean isChecksumStillValid(String url, UrlVersion urlVersion, O
    */
   private boolean isValidDownload(String url, String tool, String version, HttpResponse<?> response) {
 
+    String contentType = response.headers().firstValue("content-type").orElse("undefined");;
     if (isSuccess(response)) {
-      String contentType = response.headers().firstValue("content-type").orElse("undefined");
-      if (contentType.startsWith("text")) {
-        logger.error("For tool {} and version {} the download has an invalid content type {} for URL {}", tool, version,
-            contentType, url);
-        return false;
-      }
+      return isValidContentType(contentType);
     } else {
+      logger.error("For tool {} and version {} the download has an invalid content type {} for URL {}", tool, version,
```

Comment:
This log message is wrong/missleading.
It will happen if `isSuccess(response)` returned `false` so statusCode is not 200/2xx.
If you did not want to keep the log message in `isValidContentType` you would need to move it into the block before the `else`. And you could even keep the `contentType` declaration inside that block.

Reply:
Thanks for your suggestion, I've moved it now.